webui/react/src/stores/auth.ts

```diff
@@ -20,31 +20,46 @@ const ensureAuthCookieSet = (token: string): void => {
   if (!getCookie(AUTH_COOKIE_KEY)) setCookie(AUTH_COOKIE_KEY, token);
 };
 
-const internalAuth = new WritableObservable<Loadable<Auth>>(NotLoaded);
-const internalAuthChecked = new WritableObservable(false);
+interface AuthState {
+  auth: Loadable<Auth>;
+  isChecked: boolean;
+}
 
-export const auth = internalAuth.readOnly();
-export const authChecked = internalAuthChecked.readOnly();
+const defaultState: AuthState = {
+  auth: NotLoaded,
+  isChecked: false,
+};
 
-export const reset = (): void => {
-  clearAuthCookie();
-  globalStorage.removeAuthToken();
-  WritableObservable.batch(() => {
-    internalAuth.set(NotLoaded);
-    internalAuthChecked.set(false);
+class AuthStore {
+  protected _$state: WritableObservable<AuthState> = observable(defaultState);
+  public readonly $auth = this._$state.select((s) => s.auth);
```

Comment:
What's with the `$`?

Reply:
yeah... I don't like it after using a while. The original thinking was that it would be easy to distinguish it as an observable if it has a `$` prefix.